### ISSUE TITLE
`queueTest` has a race condition.

### ISSUE THREAD

**Issue by j-stephan** [alpaka-group/alpaka#2007]
Under random circumstances `queueTest` will fail with the following error message:

```
38: /home/runner/work/alpaka/alpaka/test/unit/queue/src/QueueTest.cpp:101: FAILED:
38:   CHECK( alpaka::empty(f.m_queue) )
38: with expansion:
38:   false
```

Example: https://github.com/alpaka-group/alpaka/actions/runs/5620806464/job/15230414419?pr=2006#step:5:7252

This is probably related to #1741 and #1936.

**Comment by j-stephan** [comment]
Ping @bernhardmgruber @psychocoderHPC

**Comment by j-stephan** [comment]
Also happens on Windows: https://github.com/alpaka-group/alpaka/actions/runs/5622448322/job/15235142056?pr=2006

**Comment by fwyzard** [comment]
It seems to be failing fairly consistently in the debug builds.

**Comment by psychocoderHPC** [comment]
The last data race should be solved with #2011
